src/tribler/core/start_core.py

```diff
@@ -169,6 +169,7 @@ def run_tribler_core_session(api_port: int, api_key: str, state_dir: Path, gui_t
     try:
         exit_code = loop.run_until_complete(core_session(config, components=list(components_gen(config))))
     finally:
+        logger.info(f'Finished with code: {exit_code if exit_code else 0}')
```

Comment:
This is not a part of the printing logic and was added during the investigation of #7489.
I can extend this change to another PR if necessary.

Reply:
This line looks incorrect for two reasons:
1. In case of an exception in `loop.run_until_complete`, this line will lead to `UnboundLocalError: local variable 'exit_code' referenced before assignment`.
2. In case of exception in `loop.run_until_complete`, it is misleading to say that the process is finished with exit code 0, because 0 means that the process is finished correctly and without errors.

As this line does not relate to other changes in this PR, I think it can be removed. And to make the line correct, you may initialize the `exit_code` variable to None before the `try/finally` block and output `The asyncio loop is finished with exception` in case the `exit_code` is still None